pkg/controller/ephemeralcluster/reconciler.go

```diff
@@ -0,0 +1,247 @@
+	if err != nil {
+		upsertProvisioningCond(ephemeralclusterv1.ConditionFalse, ephemeralclusterv1.CIOperatorJobsGenerateFailure, err.Error())
+		return
+	}
+
+	location, err := r.uploadCIOperatorConfig(ctx, ciOperatorConfig)
+	if err != nil {
+		err := fmt.Errorf("upload ci config: %w", err)
+		upsertProvisioningCond(ephemeralclusterv1.ConditionFalse, ephemeralclusterv1.CIOperatorJobsGenerateFailure, err.Error())
+		return
+	}
+	r.logger.WithField("Path", location).Info("Config uploaded to GCS")
+
+	if len(pj.Spec.PodSpec.Containers) != 1 {
+		upsertProvisioningCond(ephemeralclusterv1.ConditionFalse, ephemeralclusterv1.CIOperatorJobsGenerateFailure, "too many presubmit containers")
+		return
+	}
+	container := &pj.Spec.PodSpec.Containers[0]
+	container.Env = append(container.Env, corev1.EnvVar{Name: "CONFIG_SPEC_GCS_URL", Value: location})
+
+	if err := r.masterClient.Create(ctx, pj); err != nil {
+		err = fmt.Errorf("create prowjob: %w", err)
+		upsertProvisioningCond(ephemeralclusterv1.ConditionFalse, ephemeralclusterv1.CIOperatorJobsGenerateFailure, err.Error())
+		return
+	}
+
+	ec.Status.ProwJobID = pj.Name
+	upsertProvisioningCond(ephemeralclusterv1.ConditionTrue, "", "")
+}
+
+func (r *reconciler) makeProwJob(ciOperatorConfig *api.ReleaseBuildConfiguration) (*prowv1.ProwJob, error) {
+	jobConfig, err := prowgen.GenerateJobs(ciOperatorConfig, &prowgen.ProwgenInfo{
+		Metadata: api.Metadata{
+			Org:    "org",
+			Repo:   "repo",
+			Branch: "branch",
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("generate jobs: %w", err)
+	}
+
+	orgRepo := ciOperatorConfig.Metadata.Org + "/" + ciOperatorConfig.Metadata.Repo
+	presubs, ok := jobConfig.PresubmitsStatic[orgRepo]
+	if !ok {
+		return nil, errors.New("no presubmits generated")
+	}
+
+	var presubmit *prowconfig.Presubmit
+	for i := range len(presubs) {
+		p := &presubs[i]
+		if p.Name == "pull-ci-org-repo-branch-cluster-provisioning" {
+			presubmit = p
+			break
+		}
+	}
+	if presubmit == nil {
+		return nil, errors.New("presubmit job not found")
+	}
+
+	pj := r.newPresubmit(github.PullRequest{}, "fake", *presubmit, "no-event-guid", nil, pjutil.RequireScheduling(true))
+	pj.Spec.Refs = nil
+	pj.Spec.Report = false
+
+	return &pj, nil
+}
+
+func (r *reconciler) uploadCIOperatorConfig(ctx context.Context, config *api.ReleaseBuildConfiguration) (string, error) {
+	gcsPath := "ephemeral-cluster/configs"
+	configBytes, err := yaml.Marshal(config)
+	if err != nil {
+		return "", fmt.Errorf("marshal config: %w", err)
+	}
+	return r.configUploader.UploadConfigSpec(ctx, gcsPath, string(configBytes))
+}
+
+func (r *reconciler) upsertCondition(ec *ephemeralclusterv1.EphemeralCluster, t ephemeralclusterv1.EphemeralClusterConditionType, status ephemeralclusterv1.ConditionStatus, reason, msg string) {
+	newCond := ephemeralclusterv1.EphemeralClusterCondition{
+		Type:               t,
+		Status:             status,
+		LastTransitionTime: v1.NewTime(r.now()),
```

Comment:
Is this important for testing? You can just skip this comparison.

Reply:
Actually I'm using our testing functions [/pkg/testhelper/testhelper.go](https://github.com/openshift/ci-tools/blob/90c38df4313882f353b079bc8369a048bd49b546/pkg/testhelper/testhelper.go) and fields can't be skipped.

Reply:
You should be able to update the helper functions to ignore specific fields. If its complicated, feel free to do it in a follow-up.